config/api.yml

```diff
@@ -1134,6 +1134,56 @@
       :get:
       - :name: read
         :identifier: metrics_view
+  :middleware_datasources:
+    :description: Middleware Datasources
+    :klass: MiddlewareDatasource
+    :options:
+      - :collection
+    :verbs: *g
+    :collection_actions:
+      :get:
+        - :name: read
+          :identifier: middleware_datasource_show_list
+  :middleware_deployments:
+    :description: Middleware Deployments
+    :klass: MiddlewareDeployment
+    :options:
+      - :collection
+    :verbs: *g
+    :collection_actions:
+      :get:
+        - :name: read
+          :identifier: middleware_deployment_show_list
+  :middleware_domains:
+    :description: Middleware Domains
+    :klass: MiddlewareDomain
+    :options:
+      - :collection
+    :verbs: *g
+    :collection_actions:
+      :get:
+        - :name: read
+          :identifier: middleware_domain_show_list
+  :middleware_messagings:
+    :description: Middleware Messagings
+    :klass: MiddlewareMessaging
+    :options:
+      - :collection
+    :verbs: *g
+    :collection_actions:
+      :get:
+        - :name: read
+          :identifier: middleware_messaging_show_list
+  :middleware_servers:
+    :description: Middleware Servers
+    :klass: MiddlewareServer
+    :options:
+      - :collection
+    :verbs: *g
+    :collection_actions:
+      :get:
+        - :name: read
+          :identifier: middleware_server_show_list
```

Comment:
the new collections above need resource specific actions, for GET's they have different role identifiers (the _show ones instead of the _show_list from the miq_product_features.yml file)>

Reply:
we also need to support the query action on the new collections.